src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -191,14 +197,73 @@ public DefineFunctionsResult DefineFunctions(string script, bool numberIsFloat =
             var udfDefinitions = result.UDFs.Select(udf => new UDFDefinition(
                 udf.Ident.ToString(),
                 new ParseResult(udf.Body, errors, result.HasError, comments, null, null, script),
-                udf.ReturnType.GetFormulaType(),
+                GetFormulaTypeFromName(udf.ReturnType._value),
                 udf.IsImperative,
                 udf.NumberIsFloat,
-                udf.Args.Select(arg => new NamedFormulaType(arg.NameIdent.ToString(), arg.TypeIdent.GetFormulaType())).ToArray())).ToArray();
+                udf.Args.Select(arg => new NamedFormulaType(arg.NameIdent.ToString(), GetFormulaTypeFromName(arg.TypeIdent._value))).ToArray())).ToArray();
 
             return DefineFunctions(udfDefinitions);
         }
 
+        public void DefineType(string script, ParserOptions parserOptions)

```

Comment:
>string script [](http://example.com/codeflow?start=31&length=13)

can we clarify - which grammar is this script? 

an expression?
a type definition?

a UDF?